src/google/protobuf/util/internal/default_value_objectwriter.cc

```diff
@@ -156,8 +153,10 @@ DefaultValueObjectWriter* DefaultValueObjectWriter::RenderString(
   } else {
     // Since StringPiece is essentially a pointer, takes a copy of "value" to
     // avoid ownership issues.
-    string_values_.push_back(new string(value));
-    RenderDataPiece(name, DataPiece(*string_values_.back(), true));
+    // To avoid unnecessary call of std::string() constructor, make sure that
+    // StringPiece data is valid and pass only pointer
+    string_values_.push_back(value.empty() ? value : value.data());
```

Comment:
I think you can just call `string_values_.emplace_back(value);` as a way to minimize the number of unnecessary string constructions.

Reply:
Ok, and yeah, it's much simpler)